locust/test/test_fasthttp.py

```diff
@@ -0,0 +1,389 @@
+        r = s.options("/request_method")
+        self.assertEqual(200, r.status_code)
+        self.assertEqual("", r.content.decode())
+        self.assertEqual(
+            set(["OPTIONS", "DELETE", "PUT", "GET", "POST", "HEAD", "PATCH"]),
+            set(r.headers["allow"].split(", ")),
+        )
+
+
+class TestRequestStatsWithWebserver(WebserverTestCase):
+    def test_request_stats_content_length(self):
+        class MyLocust(FastHttpLocust):
+            host = "http://127.0.0.1:%i" % self.port
+    
+        locust = MyLocust()
+        locust.client.get("/ultra_fast")
+        self.assertEqual(global_stats.get("/ultra_fast", "GET").avg_content_length, len("This is an ultra fast response"))
+        locust.client.get("/ultra_fast")
+        self.assertEqual(global_stats.get("/ultra_fast", "GET").avg_content_length, len("This is an ultra fast response"))
+    
+    def test_request_stats_no_content_length(self):
+        class MyLocust(FastHttpLocust):
+            host = "http://127.0.0.1:%i" % self.port
+        l = MyLocust()
+        path = "/no_content_length"
+        r = l.client.get(path)
+        self.assertEqual(global_stats.get(path, "GET").avg_content_length, len("This response does not have content-length in the header"))
+    
+    def test_request_stats_no_content_length_streaming(self):
+        class MyLocust(FastHttpLocust):
+            host = "http://127.0.0.1:%i" % self.port
+        l = MyLocust()
+        path = "/no_content_length"
+        r = l.client.get(path, stream=True)
+        self.assertEqual(0, global_stats.get(path, "GET").avg_content_length)
+    
+    def test_request_stats_named_endpoint(self):
+        class MyLocust(FastHttpLocust):
+            host = "http://127.0.0.1:%i" % self.port
+    
+        locust = MyLocust()
+        locust.client.get("/ultra_fast", name="my_custom_name")
+        self.assertEqual(1, global_stats.get("my_custom_name", "GET").num_requests)
+    
+    def test_request_stats_query_variables(self):
+        class MyLocust(FastHttpLocust):
+            host = "http://127.0.0.1:%i" % self.port
+    
+        locust = MyLocust()
+        locust.client.get("/ultra_fast?query=1")
+        self.assertEqual(1, global_stats.get("/ultra_fast?query=1", "GET").num_requests)
+    
+    def test_request_stats_put(self):
+        class MyLocust(FastHttpLocust):
+            host = "http://127.0.0.1:%i" % self.port
+    
+        locust = MyLocust()
+        locust.client.put("/put")
+        self.assertEqual(1, global_stats.get("/put", "PUT").num_requests)
+    
+    def test_request_connection_error(self):
+        class MyLocust(FastHttpLocust):
+            host = "http://localhost:1"
+        
+        locust = MyLocust()
+        response = locust.client.get("/", timeout=0.1)
+        self.assertEqual(response.status_code, 0)
+        self.assertEqual(1, global_stats.get("/", "GET").num_failures)
+        self.assertEqual(0, global_stats.get("/", "GET").num_requests)
```

Comment:
Since 0.10.0 (see #939), both successful and failed requests are counted in total number of requests.
```suggestion
        self.assertEqual(1, global_stats.get("/", "GET").num_requests)
```
I really don't mean to be rude, but can we get this merged any time soon? Are there any more issues that need to be solved, I'd be happy to take a look in that case.

Cheers!